cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
Do we expect `--no-resolve-image` to persist? I expect that it would only apply to the current deploy.

For any service with a missing label we can treat that as an image name mismatch. The first deploy will cause an update to include the service image label. 

Otherwise I wouldn't expect `--no-resolve-image` to change the label value (it will always contain the string from the config, not a digest), so the flag acts as an override.